shared/js/media/video_player.js

```diff
@@ -161,6 +167,15 @@ function VideoPlayer(container) {
     // The only place we call showPlayer() is from the play() function.
     // If play() has to show the player, call it again when we're ready to play.
     player.oncanplay = function() {
+      if (videowidth != player.videoWidth ||
+          videoheight != player.videoHeight)
+      {
+        // Our guess at the video size was wrong, resize the display
```

Comment:
Does this actually happen in practice? Your gecko code creates a poster that is screen sized even when the video is larger than the screen? That seems like a reasonable thing to do. But I would have thought that the simplest thing would be to just get a frame and encode it at full size.

I'll want to wait to try this out in practice to see if it causes flicker or anything.  I suspect that Gallery has a hard-coded assumption that the poster size matches the video size, and if so, we'd hit this every time. So I might also want to modify gallery to get and save the real size.

If you can make gecko guarantee that the sizes match that would simplify things...